service/provider/src/main/java/gov/va/vro/service/provider/camel/MasIntegrationRoutes.java

```diff
@@ -178,52 +180,72 @@ private void configureMasProcessing() {
-        // Check and update claim
+        .doCatch(BipException.class)
+        // Completion code needs the MasProcessingObject as the body.
+        .setBody(simple("${exchangeProperty.payload}"))
+        .setProperty("offRampError", constant(pdfFailError))
+        .setProperty("sourceRoute", constant(rfdRouteId))
+        .to(ENDPOINT_OFFRAMP_ERROR)
+        .stop()
+        .end() // End try
         .to(ENDPOINT_MAS_COMPLETE);
 
     // Call "Order Exam" in the absence of evidence .i.e Sufficient For Fast Tracking is "false"
     var orderExamRouteId = "mas-order-exam";
+    final String orderFailMessage = "examOrderFailed";
+    final String pdfFailMessage = "PDF upload failed after exam order requested.";
+
     from(ENDPOINT_ORDER_EXAM)
         // input: MasAutomatedClaimPayload
         .routeId(orderExamRouteId)
         .wireTap(ENDPOINT_AUDIT_WIRETAP)
         .onPrepare(
             auditProcessor(orderExamRouteId, "There is insufficient evidence. Ordering an exam"))
+        .doTry()
         .process(masOrderExamProcessor)
         .log("MAS Order Exam response: ${body}")
         // Upload PDF but catch errors since exam was ordered and continue
-        .doTry()
         .to(ENDPOINT_UPLOAD_PDF)
         .to(ENDPOINT_MAS_COMPLETE)
+        .doCatch(MasException.class)
+        // Body is still the Mas Processing object.
+        .setProperty("sourceRoute", constant(orderExamRouteId))
+        .setProperty("offRampError", constant(orderFailMessage))
+        .to(ENDPOINT_OFFRAMP_ERROR)
+        .stop() // Offramp and don't continue processing
         .doCatch(BipException.class)
         // Mas Complete Processing code expects this to be the body of the message
         .setBody(simple("${exchangeProperty.payload}"))
+        // Wiretap will cause no code to execute after the end of the try. Intentional here.
         .wireTap(ENDPOINT_NOTIFY_AUDIT) // Send error notification to slack
-        .onPrepare(
-            slackEventProcessor(orderExamRouteId, "PDF upload failed after exam order requested."))
+        .onPrepare(slackEventProcessor(orderExamRouteId, pdfFailMessage))
         .to(ENDPOINT_MAS_COMPLETE)
-        .endDoCatch();
+        .end();
 
-    // Off Ramp if the Sufficiency can't be determined .i.e. sufficientForFastTracking is 'null'
-    var assessorErrorRouteId = "assessorError";
-    from(ENDPOINT_ACCESS_ERR)
-        .routeId(assessorErrorRouteId)
-        .log("Assessor Error. Off-ramping claim")
-        .process(masAccessErrProcessor)
-        .wireTap(ENDPOINT_NOTIFY_AUDIT)
-        .onPrepare(slackEventProcessor(assessorErrorRouteId, "Sufficiency cannot be determined."))
-        .to(ENDPOINT_MAS_COMPLETE);
+    // Wiretap does NOT let camel work as expected when placed directly inside doCatch()
+    // Thus it is broken out here, in the interest of letting normal flow/control happen.
+    from(ENDPOINT_OFFRAMP_ERROR)
+        .wireTap(ENDPOINT_NOTIFY_AUDIT) // Send error notification to slack
```

Comment:
I realize I often call stop() explicitly rather than relying on this strange behavior. I think that would be easier for devs to follow in the future. And if camel ever *makes that work* the way that it seems it should.

Reply:
Not quite but I will sync with you in a huddle to explain. :) easier i think